test/integration/scheduler/queue_test.go

```diff
@@ -336,6 +371,132 @@ func TestCoreResourceEnqueue(t *testing.T) {
 	}
 }
 
+// TestCoreResourceNoEnqueue verify Pods failed by in-tree default plugins
+// will not be moved to active queue by irrelevant events.
+func TestCoreResourceNoEnqueue(t *testing.T) {
+	tests := []struct {
+		name string
+		// initialNode is the Node to be created at first.
+		initialNode *v1.Node
+		// initialPod is the Pod to be created at first if it's not empty.
+		initialPod *v1.Pod
+		// pods are the list of Pods to be created.
+		// All of them are expected to be unschedulable at first.
+		pods []*v1.Pod
+		// triggerFn is the function that triggers the event to move Pods.
+		triggerFn func(testCtx *testutils.TestContext) error
+	}{
+		{
+			name:        "Updating pod's condition doesn't make it requeued to activeQ",
+			initialNode: st.MakeNode().Name("fake-node").Taints([]v1.Taint{{Key: v1.TaintNodeNotReady, Effect: v1.TaintEffectNoSchedule}}).Obj(),
+			pods: []*v1.Pod{
+				// - Pod1 doesn't have the required toleration and will be rejected by the TaintToleration plugin.
+				st.MakePod().Name("pod1").Container("image").Obj(),
+			},
+			triggerFn: func(testCtx *testutils.TestContext) (err error) {
+				// Trigger a Pod Condition update event.
```

Comment:
Let's leave a TODO comment:

```go
// Simulate a Pod update by directly calling `SchedulingQueue.Update` instead of actually updating a Pod
// because we don't have a way to confirm the scheduler has handled a Pod update event at the moment.
// TODO: actually update a Pod update and confirm the scheduler has handled a Pod update event with a metric.
// https://github.com/kubernetes/kubernetes/pull/122234#discussion_r1597456808
```

Reply:
Added.